cdap-ui/app/cdap/components/Replicator/Create/Content/TargetConfig/index.tsx

```diff
@@ -42,15 +44,33 @@ const TargetConfigView: React.FC<ICreateContext & WithStyles<typeof styles>> = (
   setTargetConfig,
 }) => {
   const [values, setValues] = React.useState(targetConfig);
+  const [loading, setLoading] = React.useState(false);
 
-  // Fetch Target
   React.useEffect(() => {
-    // TODO: replace with data from Target List
-    const artifactName = 'delta-bigquery-plugins';
-    const artifactScope = 'SYSTEM';
-    const pluginName = 'bigquery';
+    if (targetPluginWidget || !targetPluginInfo) {
+      return;
+    }
+
+    const artifact = targetPluginInfo.artifact;
+
+    fetchPluginWidget(
+      artifact.name,
```

Comment:
Are there any types we already have or could add for artifacts and plugins?

Reply:
Actually I see you do have types below in utilities - can we use them here?

Reply:
types as in an interface? this is a function arguments signature, so all of these properties are string and expected to be in certain order

Reply:
Right! I think I meant interfaces for plugins and artifacts (and you are right that it wouldn't actually change anything here, my bad). You addressed the question of adding interfaces in another PR, I think, so totally fine to ignore this rather confusing and poorly worded comment.